foyer/forcefield.py

```diff
@@ -340,18 +345,26 @@ def createSystem(self, topology, atomtype=True, nonbondedMethod=NoCutoff,
             the newly created System
         """
         if atomtype:
-            residue_map = dict()
-
-            for res in topology.residues():
-                if res.name not in residue_map.keys():
-                    residue = _topology_from_residue(res)
-                    find_atomtypes(residue, forcefield=self)
-                    residue_map[res.name] = residue
-        
-            for key, val in residue_map.items():
-                new_system = _update_atomtypes(topology, val)
-        
-            topology = new_system
+            independent_residues = _check_independent_residues(topology)
+
+            if independent_residues == True:
```

Comment:
In general, when checking if a boolean value is True or False, it is better to just check like so:
```python
if independent_residues:
   ...
```